workshops/geometric_pattern/README.md

```diff
@@ -0,0 +1,538 @@
+    }
+    alternatingRow = !alternatingRow;
+  }
+}
+```
+
+This sets the flag `alternatingRow` to its opposite. Its initial value was `false`, and after one row, its value will be `true`. Thus, the second row will be an alternating row. After the second row is created, the flag will be flipped back to false, and so on.
+
+Let's create a conditional that will use the flag's value to determine whether or not to shift the row. We'll add this just before the inner for-loop:
+
+```js
+function draw() {
+  var alternatingRow = false;
+  for (var y = 0; y <= height; y = y+CIRCLE_RAD) {
+    if (alternatingRow) {
+
+    } else {
+
+    }
+    for (var x = 0; x <= width; x = x+CIRCLE_DIA) {
+      ellipse(x, 0, CIRCLE_DIA, CIRCLE_DIA);
+    }
+    alternatingRow = !alternatingRow;
+  }
+}
+```
+
+Now we just need to make this shift. We'll do this by modifying the start `x` value in our inner for-loop. Currently, the start `x` value is always 0, which results in every row starting at the x-coordinate 0.
+
+We can create a variable (`startX`) within `draw()` and use our conditional to set the value of `startX` to 0 or `CIRCLE_RAD`, depending on if it's an alternating row or not. Don't forget to modify the inner for-loop so that the value of `x` starts at `startX`.
+
+```js
+function draw() {
+  var alternatingRow = false;
+  var startX;
+  for (var y = 0; y <= height; y = y+CIRCLE_RAD) {
+    if (alternatingRow) {
+      startX = CIRCLE_RAD;
+    } else {
+      startX = 0;
+    }
+    for (var x = startX; x <= width; x = x+CIRCLE_DIA) {
+      ellipse(x, 0, CIRCLE_DIA, CIRCLE_DIA);
+    }
+    alternatingRow = !alternatingRow;
+  }
+}
+```
+
+Save and check it out!
+
+### Reversing Direction
+
+So you may have noticed that it seems that our circles are overlapping on the wrong side. We wanted them to overlap each other on the top, but they are overlapping on the bottom. This is because they are being drawn from top to bottom, and the upper circles are being drawn over.
+
+We can fix this by swapping the start and end values of `y` in the outer for-loop and instead of incrementing the value of `y`, we'll decrement it. Don't forget to change the condition in the for-loop to be greater than or equal to 0 instead of less than or equal to.
+
+```js
+function draw() {
+  var alternatingRow = false;
+  var startX;
+  for (var y = height; y >= 0; y = y-CIRCLE_RAD) {
+    if (alternatingRow) {
+      startX = CIRCLE_RAD;
+    } else {
+      startX = 0;
+    }
+    for (var x = startX; x <= width; x = x+CIRCLE_DIA) {
+      ellipse(x, 0, CIRCLE_DIA, CIRCLE_DIA);
+    }
+    alternatingRow = !alternatingRow;
+  }
+}
+```
+
+Save to see the changes!
+
+## Part IV: Adding Color
+
+Black and white is nice, but how about some color? p5.js provides several helpful functions, including: [`color()`](http://p5js.org/reference/#p5/color), [`fill()`](http://p5js.org/reference/#p5/fill), and [`stroke()`](http://p5js.org/reference/#p5/stroke) that help modify the graphics.
```

Comment:
Don't think anyone will wonder too much since it's pretty obvious after the next step where you actually use the functions.